### PR TITLE
[Feature] support display cloud native pk table index usage in show cmd

### PR DESCRIPTION
**Pull request by luohaha**
Why I'm doing:
We need to display each cloud native pk tablet's memory and disk usage about primary index.

What I'm doing:
support display cloud native pk table index usage in show cmd, e.g:
```
mysql> show tablet from tbl_pk_novarchar;
+----------+-----------+----------+-----------+-----------------+------------------+
| TabletId | BackendId | DataSize | RowCount  | PrimaryIndexMem | PrimaryIndexDisk |
+----------+-----------+----------+-----------+-----------------+------------------+
| 11005    | [10004]   | 10.2GB   | 191996368 | 262MB           | 5.4GB            |
+----------+-----------+----------+-----------+-----------------+------------------+
1 row in set (0.00 sec)
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by sduzh** [comment]
Hi @luohaha, I think it's essential to have statistics on tablet's memory and disk usage, but whether or not to get it via show proc is something I think there's a discussion about, and these are some of my concerns:
1. Each tablet fetching disk space requires traversing the directory once, has there been any evaluation of what additional cost is introduced (IO, latency) and whether it is acceptable.
2. Under multi-warehouse, a Tablet may exist on multiple nodes, how to choose and control which node to get the stats on?
3. Tablet memory usage may change at any time, but show proc gets the result periodically, and the result of show proc may be collected a long time ago, so the result of show proc may be quite different from the actual situation when the command is executed, should we get the memory size through another real-time interface instead of show proc? Should we get the memory size through another real-time interface instead of show proc?

**Comment by sduzh** [comment]
Maybe we can get tablet stats by adding a system table like `information_schema.be_cloud_native_compactions`.

File: fe/fe-core/src/main/java/com/starrocks/common/proc/LakeTabletsProcDir.java [hunk 1]
```diff
@@ -42,6 +41,7 @@
 public class LakeTabletsProcDir implements ProcDirInterface {
     public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
             .add("TabletId").add("BackendId").add("DataSize").add("RowCount")
+            .add("PrimaryIndexMem").add("PrimaryIndexDisk")
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/LakeTabletsProcDir.java:44]: I'm curious what't the difference between this file and ShowExecutor.java

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java [hunk 2]
```diff
@@ -55,13 +56,19 @@ public class LakeTablet extends Tablet {
...
+    private static final String JSON_KEY_PK_INDEX_MEM_SIZE = "primaryIndexMem";
+    private static final String JSON_KEY_PK_INDEX_DISK_SIZE = "primaryIndexDisk";
 
     @SerializedName(value = JSON_KEY_DATA_SIZE)
     private volatile long dataSize = 0L;
     @SerializedName(value = JSON_KEY_ROW_COUNT)
     private volatile long rowCount = 0L;
     @SerializedName(value = JSON_KEY_DATA_SIZE_UPDATE_TIME)
     private volatile long dataSizeUpdateTime = 0L;
+    @SerializedName(value = JSON_KEY_PK_INDEX_MEM_SIZE)
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java:68]: In my opinion, the primary index size is also useful for normal tablets (LocalTablets), isn't it?

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java [hunk 3]
```diff
@@ -55,13 +56,19 @@ public class LakeTablet extends Tablet {
     private static final String JSON_KEY_DATA_SIZE = "dataSize";
     private static final String JSON_KEY_ROW_COUNT = "rowCount";
     private static final String JSON_KEY_DATA_SIZE_UPDATE_TIME = "dataSizeUpdateTime";
+    private static final String JSON_KEY_PK_INDEX_MEM_SIZE = "primaryIndexMem";
+    private static final String JSON_KEY_PK_INDEX_DISK_SIZE = "primaryIndexDisk";
 
     @SerializedName(value = JSON_KEY_DATA_SIZE)
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/lake/LakeTablet.java:62]: I'm curious that, in Database class, the serialized names are single character, but in Tablet class (like LocalTablet), the serialzed names are full words, will it affect the meta image size a lot?

**Pull request closed without merging**